test/widgets/message_list_test.dart

```diff
@@ -194,22 +199,121 @@ void main() {
+    group('StreamMessageRecipientHeader', () {
+      final stream = eg.stream(name: 'stream name');
+      final message = eg.streamMessage(stream: stream, topic: 'topic name');
+      FinderResult findInMessageList(String text) {
+        // Stream name shows up in [AppBar] so need to avoid matching that
+        return find.descendant(
+          of: find.byType(MessageList),
+          matching: find.text(text)).evaluate();
+      }
+
+      testWidgets('show stream name in AllMessagesNarrow', (tester) async {
+        await setupMessageListPage(tester,
+          narrow: const AllMessagesNarrow(),
+          messages: [message], streams: [stream]);
+        await tester.pump();
+        check(findInMessageList('stream name')).length.equals(1);
+        check(findInMessageList('topic name')).length.equals(1);
+      });
+
+      testWidgets('do not show stream name in StreamNarrow', (tester) async {
+        await setupMessageListPage(tester,
+          narrow: StreamNarrow(stream.streamId),
+          messages: [message], streams: [stream]);
+        await tester.pump();
+        check(findInMessageList('stream name')).length.equals(0);
+        check(findInMessageList('topic name')).length.equals(1);
+      });
+
+      testWidgets('do not show stream name in TopicNarrow', (tester) async {
+        await setupMessageListPage(tester,
+          narrow: TopicNarrow.ofMessage(message),
+          messages: [message], streams: [stream]);
+        await tester.pump();
+        check(findInMessageList('stream name')).length.equals(0);
+        check(findInMessageList('topic name')).length.equals(1);
+      });
+
+      testWidgets('color of recipient header background', (tester) async {
+        final subscription = eg.subscription(stream, color: Colors.red.value);
+        final swatch = subscription.colorSwatch();
+        await setupMessageListPage(tester,
+          messages: [message], subscriptions: [subscription]);
+        await tester.pump();
+        check(tester.widget<ColoredBox>(
+          find.descendant(
+            of: find.byType(StreamMessageRecipientHeader),
+            matching: find.byType(ColoredBox),
+        ))).color.equals(swatch.barBackground);
+      });
+
+      testWidgets('color of stream icon', (tester) async {
+        final stream = eg.stream(isWebPublic: true);
+        final subscription = eg.subscription(stream, color: Colors.red.value);
+        final swatch = subscription.colorSwatch();
+        await setupMessageListPage(tester,
+          messages: [message], subscriptions: [subscription]);
+        await tester.pump();
+        check(tester.widget<Icon>(find.byIcon(ZulipIcons.globe)))
+          .color.equals(swatch.iconOnBarBackground);
+      });
+
+      testWidgets('normal streams show hash icon', (tester) async {
+        final stream = eg.stream(isWebPublic: false, inviteOnly: false);
+        await setupMessageListPage(tester,
+          messages: [message], streams: [stream]);
```

Comment:
This test is fragile because it relies on `message` being to the stream `stream`, and that's true only by accident. (It's true only because `eg.stream` currently uses a constant stream ID every time by default, whereas we ought to instead make it generate a distinct random ID every time.)

Can instead write `messages: [eg.streamMessage(stream: stream)]`.